Core/NsvSlicedBlock.cs

```diff
@@ -0,0 +1,382 @@
+                    var sliceTransform = sliceGO.AddComponent<RectTransform>();
+                    slice.material = assetLoader.UINoGlowMaterial;
+                    slice.sprite = assetLoader.White;
+                    slice.color = _config.SliceColor;
+                    // slice.sortingLayerID = SortingLayerID;
+                    slice.sortingOrder = 0;
+                    // slice.maskInteraction = SpriteMaskInteraction.VisibleInsideMask;
+                    sliceTransform.SetParent(sliceGroupTransform);
+                    sliceTransform.localScale = new Vector3(_config.SliceWidth, 1f, 1f);
+                    sliceTransform.localRotation = Quaternion.identity;
+                    sliceTransform.localPosition = new Vector3(-_config.SliceWidth / 2f, -0.5f, 0f);
+                    _sliceTransform = sliceTransform;
+                    _slice = slice;
+                }
+            }
+        }
+
+        internal void Init(RectTransform parent)
+        {
+            gameObject.GetComponent<RectTransform>().SetParent(parent);
+        }
+
+        internal void SetData(NoteController noteController, NoteCutInfo noteCutInfo, NoteData noteData)
+        {
+            // Extract cube rotation from actual cube rotation
+            var cubeRotation = _config.RotationFromCubeTransform
+                ? noteController.noteTransform.localRotation.eulerAngles.z
+                : noteData.cutDirection.RotationAngle();
+            // Using built-in rotationAngle conversion.
+            // Please note that it's range is from [-180° <=> 180°[ compared to [0° <=> 360°[, but that shouldn't make a difference.
+
+            SetCubeState(noteController, noteCutInfo, noteData, cubeRotation);
+
+            SetSliceState(noteController, noteCutInfo, cubeRotation);
+        }
+
+        private void SetCubeState(NoteController noteController, NoteCutInfo noteCutInfo, NoteData noteData, float cubeRotation)
+        {
+            float cubeX;
+            float cubeY;
+            if (_config.PositionFromCubeTransform)
+            {
+                const float positionScaling = 2.0f;
+                var position = noteController.noteTransform.position * positionScaling;
+                cubeX = position.x;
+                cubeY = position.y;
+            }
+            else
+            {
+                var positionOffset = new Vector2(-1.5f, 1.5f);
+                cubeX = noteData.lineIndex + positionOffset.x;
+                cubeY = (int) noteData.noteLineLayer + positionOffset.y;
+            }
+
+            _aliveTime = 0f;
+
+            _isDirectional = noteData.cutDirection != NoteCutDirection.Any && noteData.cutDirection != NoteCutDirection.None;
+
+            if (_config.UseCustomColors)
+            {
+                _color = noteData.colorType == ColorType.ColorA ? _config.LeftColor : _config.RightColor;
+                _saberColor = noteData.colorType != ColorType.ColorA ? _config.LeftColor : _config.RightColor;
```

Comment:
```suggestion
                _saberColor = noteData.saberType == ColorType.ColorA ? _config.LeftColor : _config.RightColor;
```

Reply:
Won't compile... but I get the point.
That being said, I'm actually not sure how this will react to left-handed mode. 🤔

Reply:
I just checked it -- left-handed mode does two things:
1. mirror the notes and swap their color
2. use left saber in single saber mode
So this change works as intended.